yt_dlp/extractor/duboku.py

```diff
@@ -129,11 +131,14 @@ def _real_extract(self, url):
         data_url = player_data.get('url')
         if not data_url:
             raise ExtractorError('Cannot find url in player_data')
-        data_from = player_data.get('from')
+        if player_data.get('encrypt') == 1:
+            data_url = urllib.parse.unquote(data_url)
+        if player_data.get('encrypt') == 2:
+            data_url = urllib.parse.unquote(base64.b64decode(data_url))
```

Comment:
Unquote only works on bytes in Python >=3.9 ([ref](<https://docs.python.org/3/library/urllib.parse.html#urllib.parse.unquote>))
```suggestion
            data_url = urllib.parse.unquote(base64.b64decode(data_url).decode('ascii'))
```